README.md

```diff
@@ -203,15 +203,22 @@ session = boto3.Session(
 
 url = 's3://clp-example-s3-bucket/example.clp.zst'
 # Using `smart_open.open` to stream the encoded CLP IR:
-with smart_open.open(url, "rb", transport_params={'client': session.client('s3')}) as istream:
+with smart_open.open(url, mode="rb", compression='disable',
+                     transport_params={'client': session.client('s3')}) as istream:
     with ClpIrStreamReader(istream, allow_incomplete_stream=True) as clp_reader:
         for log_event in clp_reader:
             # Print the log message with its timestamp properly formatted.
             print(log_event.get_formatted_message())
 ```
 
 Note:
-When `allow_incomplete_stream` is set to False (default), the reader will raise
+- Setting `compression='disable'` is necessary because smart_open by default
+infers the compression format from the file extension and decompresses the file
+automatically. However, the ClpIrStreamReader expects the input stream to be in
+the original compressed format. Disabling the automatic decompression ensures
+that the stream passed to ClpIrStreamReader remains compressed, preventing
+decompression errors.
```

Comment:
```suggestion
- Setting `compression="disable"` is necessary so that `smart_open` doesn't
undo the IR file's Zstandard compression (based on the file's extension) before
streaming it to `ClpIrStreamReader`; `ClpIrStreamReader` expects the input
stream to be Zstandard-compressed.
```

This is more of a suggestion to simplify things, but if you feel the additional info is important, feel free to disagree.

Reply:
I agree with the simplification. "`ClpIrStreamReader` expects..." carries the connotation that if the expectation isn't met, some errors might occur.